### PR TITLE
Removing inactive/timed out sessions

### PR DESCRIPTION
**Pull request by conniey**
# Description

This fixes the issue that if a timeout occurs waiting for a session to be active, the same session is returned when another call to `createSession` is made.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by anuchandy — APPROVED** [review]
awesome, thank you!

**Review by ki1729 — APPROVED** [review]

File: sdk/core/azure-core-amqp/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ### Bugs Fixed
 
+- Removing inactive session when it has timed out, so `ReactorConnection.getSession(String)` does not return the same session.
```
Comment (liukun-msft) [inline sdk/core/azure-core-amqp/CHANGELOG.md:11]: Change `Removing`  to  `Removed` ?

If there is a correctponding issue,  we can also attached.

**Review by liukun-msft — APPROVED** [review]
LGTM!🚢

File: sdk/core/azure-core-amqp/CHANGELOG.md (hunk 1 above)
Comment (conniey) [reply sdk/core/azure-core-amqp/CHANGELOG.md:11]: Thanks for the catch. There was no GitHub issue associated with it. :( I should create one though so we can redirect people if they see the same issue.

**Pull request merged**